pkg/cmd/cli/cmd/cancelbuild.go

```diff
@@ -105,7 +105,7 @@ func RunCancelBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 	// Create a new build with the same configuration.
 	if cmdutil.GetFlagBool(cmd, "restart") {
 		request := &buildapi.BuildRequest{
-			ObjectMeta: kapi.ObjectMeta{Name: build.Name},
+			ObjectMeta: kapi.ObjectMeta{Name: build.Name, Namespace: namespace},
 		}
 		newBuild, err := client.Builds(namespace).Clone(request)
```

Comment:
Namespace is special because it's already in the URL.

----- Original Message -----
> > @@ -105,7 +105,7 @@ func RunCancelBuild(f *clientcmd.Factory, out
> > io.Writer, cmd *cobra.Command, arg
> >  	// Create a new build with the same configuration.
> >  	if cmdutil.GetFlagBool(cmd, "restart") {
> >  		request := &buildapi.BuildRequest{
> > -			ObjectMeta: kapi.ObjectMeta{Name: build.Name},
> > +			ObjectMeta: kapi.ObjectMeta{Name: build.Name, Namespace: namespace},
> >  		}
> >  		newBuild, err := client.Builds(namespace).Clone(request)
> 
> Let me also say: "blech.  Making a caller submit valid resources seem
> reasonable to me"
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2735/files#r31920203